modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryMetricsTests.java

```diff
@@ -244,8 +258,74 @@ public void testMetricsForRequestRangeNotSatisfied() {
         }
     }
 
+    public void testRetrySnapshotDeleteMetricsOnEventualSuccess() throws IOException {
+        final int maxRetries = 5;
+        final String repositoryName = randomRepositoryName();
+        // Disable retries in the client for this repo
+        createRepository(
+            repositoryName,
+            Settings.builder()
+                .put(repositorySettings(repositoryName))
+                .put(S3ClientSettings.MAX_RETRIES_SETTING.getConcreteSettingForNamespace("placeholder").getKey(), 0)
```

Comment:
It seems as though putting client settings in the repository settings like this [is deprecated](https://www.elastic.co/guide/en/elasticsearch/plugins/7.17/repository-s3-repository.html), and perhaps I've done it wrong as well. Any advice on how to override AWS client internal retries on a per-test basis would be appreciated.

I tried making a different client with that setting in the node settings (set at the class level), but there are a lot of additional settings configured for the `test` client that I'd need to duplicate to make that work.

Reply:
It feels like the correct way to do this would be to change the client settings in `S3BlobStoreRepositoryTests` to be configured for `default` allowing them to be selectively overridden in child classes. But that's a larger change. Perhaps there's an easier/better way?

Reply:
I think it's ok in tests, we don't have a plan for removing this apparently-deprecated functionality any time soon.